js/app.js

```diff
@@ -131,392 +58,160 @@ app.controller('CheckoutController', function($scope, $interval, Order, $httpPar
-                alt_totalTime = 600;
-                $scope.alt_clock = data.expires;
-                $scope.alt_tick_interval = $interval($scope.alt_tick, 1000);
-                $scope.order.altsymbol = altsymbol;
-                $scope.altcoinselect = $scope.altcoins[altsymbol];
-                $scope.spinner = false;
+Url.$inject = ['$httpParamSerializer'];
+function Url($httpParamSerializer) {
+    var service = this;
 
-                process_alt_response(data);
-                    //Fetch the order id using bitcoin address
-                    Order.get({
-                        "get_order": data.order.destination
-                    }, function(order) {
-                        $scope.order.order_id = order.order_id;
-                    });
-                });
+    service.get_parameter_by_name = function(name, url) {
+        if (!url) {
+            url = window.location.href;
         }
+        name = name.replace(/[\[\]]/g, "\\$&");
+        var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+        results = regex.exec(url);
+        if (!results) return null;
+        if (!results[2]) return '';
+        return decodeURIComponent(results[2].replace(/\+/g, " "));
     }
 
-    //Process altcoin response
-    function process_alt_response(data) {
-        if( isRefundAddress() ){
-        //Refunded
-        $scope.altuuid = get_uuid();
-        $scope.altrefund = data.refund_address;
-        update_altcoin_status('refunded');
-    }
-    else if( needsRefund() ){
-        //Needs Refund
-        $scope.altuuid = get_uuid();
-        update_altcoin_status('add_refund');
-        stop_interval();
-    }
-    else if( "WAITING_FOR_DEPOSIT" == data.status ){
-        //Waiting
-        update_altcoin_status('waiting');
-        start_check_order();
-    }
-    else if( ("DEPOSIT_RECEIVED" == data.status || "DEPOSIT_CONFIRMED" == data.status || "EXECUTED" == data.status) ){
-        //Recieved
-        update_altcoin_status('received');
-        stop_interval();
-    }
-    else if( "EXPIRED" == data.status ){
-        //Expired
-        update_altcoin_status('expired');
-        stop_interval();
-    }
-
-    function gotWrongAmount(){
-        if( ["UNDERPAY_RECEIVED", "UNDERPAY_CONFIRMED", "OVERPAY_RECEIVED", "OVERPAY_CONFIRMED"].indexOf(data.payment_status) > -1 ){
-            return true;
-        }
-        return false;
-    }
-    function isPaymentReceived(){
-        if( ["PAYMENT_RECEIVED", "PAYMENT_CONFIRMED"].indexOf(data.payment_status) > -1 ){
-            return true;
-        }
-        return false;
-    }
-    function gotFunds(){
-        if(gotWrongAmount() || isPaymentReceived()){
-            return true;
+    //There are two styles of callback url in 
+    //woocommerce, we have to support both
+    //https://docs.woocommerce.com/document/wc_api-the-woocommerce-api-callback/
+    service.get_wc_endpoint = function(new_param = '', order_id = '', crypto_code = '') {
```

Comment:
This should be a generic function. It takes one dict() of parameter/value as input. Should not have specific parameters as given here (that limits the scope of this function a lot)

Reply:
Updated in 8686fdd